src/satellite_images_nso/_manipulation/nso_manipulator.py

```diff
@@ -39,17 +40,31 @@ def __make_the_crop(coordinates, raster_path, raster_path_cropped, plot):
     @param plot: Plot the results true or false
     """
 
-    geometry = [Polygon(coords) for coords in coordinates]
-    geo_area_to_use = gpd.GeoDataFrame(geometry=geometry, crs="EPSG:4326")
+    if len(coordinates) == 1:
+        geometry = [Polygon(coords) for coords in coordinates]
+        geo_area_to_use = gpd.GeoDataFrame(geometry=geometry, crs="EPSG:4326")
+
+        if geo_area_to_use.crs != "epsg:28992":
+            geo_area_to_use = geo_area_to_use.to_crs(epsg=28992)
+        area_to_crop = geo_area_to_use["geometry"]
+
+    elif len(coordinates) > 1:
+
+        print("Cropping multipolygons")
+        polygons = []
+        for x in range(0, len(coordinates)):
+            polygon = Polygon(coordinates[x][0])
+            polygons.append(polygon)
+        agdf = gpd.GeoDataFrame(geometry=polygons, crs="EPSG:4326")
+        agdf = agdf.to_crs("EPSG:28992")
+        area_to_crop = [mapping(agdf.unary_union)]

```

Comment:
I believe this line is the only functional difference between `len(coordinates) == 1` and `len(coordinates) > 1`

Reply:
No, I mean that you can move everything else out of the if/else statement. Only line 60/line 49 respectively would need to be in an if else statement.